web/src/components/HowItWorksModal.tsx

```diff
@@ -0,0 +1,169 @@
+      </>
+    ),
+    imageSrc: CreateWorkspaceButtonSVG,
+  },
+  {
+    title: "2. Paste Workspace Group's password",
+    description: (
+      <>
+        This can be obtained from the <b>Connect Directly flyout</b>. The
+        password can be reset in the Access tab of your Workspace Group.
+      </>
+    ),
+    imageSrc: WorkspaceConnectOptionSVG,
+  },
+  {
+    title: "3. Select Martech database and connect",
+    description: (
+      <>
+        Follow the <b>Try with sample data</b> tutorial in SingleStoreDB help
+        menu to know how to create and connect to a Martech database.
+      </>
+    ),
+    imageSrc: RTDMConnectButtonSVG,
+  },
+];
+
+const StepSeparator = () => (
+  <Box
+    display="inline-flex"
+    padding={3}
+    justifyContent="center"
+    alignItems="center"
+  >
+    <Icon
+      as={HiChevronRight}
+      color={useColorModeValue("white", "#2F206E")}
+      background={useColorModeValue("#553ACF", "#CCC3F9")}
+      borderRadius="50%"
+      fontSize="1.2em"
+    />
+  </Box>
+);
+
+export const HowItWorksModal = () => {
+  const [showModal, setShowModal] = React.useState(false);
+
+  const openModal = () => setShowModal(true);
+  const closeModal = () => setShowModal(false);
+
+  const SingleStepContainer = ({
+    title,
+    description,
+    imageSrc,
+  }: {
+    title: string;
+    description: React.ReactNode;
+    imageSrc: string;
+  }) => (
+    <GridItem
+      alignItems="center"
+      justifyContent="center"
+      display="flex"
+      flexDirection="column"
+      textAlign="center"
+      gap="8px"
+    >
+      <Image
+        src={imageSrc}
+        width="70%"
+        objectFit="contain"
+        marginBottom="16px"
+      />
+      <Text fontSize="md" fontWeight="bold">
+        {title}
+      </Text>
+      <Text fontSize="sm">{description}</Text>
+    </GridItem>
+  );
+
+  const StepGridItems = () => {
+    const numOfSteps = HOW_IT_WORKS_STEPS.length;
+    return (
+      <>
+        {HOW_IT_WORKS_STEPS.map(({ title, description, imageSrc }, index) => (
+          <React.Fragment key={index}>
+            <SingleStepContainer
+              title={title}
+              description={description}
+              imageSrc={imageSrc}
+            />
+            {/* We add a arrow separator in between each steps.*/}
+            {index < numOfSteps - 1 && <StepSeparator />}
+          </React.Fragment>
+        ))}
+      </>
+    );
+  };
+
+  return (
+    <>
+      <Link onClick={openModal}>How it works?</Link>
+      <Modal size="6xl" isOpen={showModal} onClose={closeModal} isCentered>
+        <ModalOverlay />
+        <ModalContent
+          display="flex"
+          alignItems="center"
+          padding="24px 26px 24px 26px"
+        >
+          <Stack justifyContent="center" alignItems="center" gap="26px">
+            <ModalHeader fontSize="26px" fontWeight="bold" padding="8px">
+              How Real-Time Digital marketing app works?
```

Comment:
nit:

```suggestion
              How does the Real-Time Digital marketing app work?
```

Reply:
Honestly, I do not like either of the statement. We are not explaining how the RTDM works rather we are explaining  how the connection to singlestoreDB works

Should we better rename it something like

"How does the connection to singlesytoreDB work in RTDM?"

does this make sense ?? @TitoGrine

Reply:
Agreeded, what do you think about:

"How to connect the Real-Time Digital marketing app to SingleStoreDB?"

Reply:
deal! It seems good to me

Reply:
![Screenshot from 2023-03-24 11-33-31](https://user-images.githubusercontent.com/20814995/227437865-52016134-42ec-49ae-87da-fcd2088da917.png)